sbnana/CAFAna/Core/MultiVar.cxx

```diff
@@ -12,11 +12,71 @@ namespace ana
   {
   }
 
+  //----------------------------------------------------------------------
+  /// Helper for \ref MultiVar2D
+  template<class T> class MultiVar2DFunc
+  {
+  public:
+    MultiVar2DFunc(const _MultiVar<T>& a, const Binning binsa,
+                   const _MultiVar<T>& b, const Binning binsb)
+      : fA(a), fBinsA(binsa),
+        fB(b), fBinsB(binsb)
+    {
+    }
+
+    std::vector<double> operator()(const T* sr) const
+    {
+      std::vector<double> returnVec;
+
+      const std::vector<double> vaVec = fA(sr);
+      const std::vector<double> vbVec = fB(sr);
+
+      if(vaVec.size() != vbVec.size())
+        throw std::length_error("MultiVars need to be same size");
+
+      for(unsigned n = 0; n < vaVec.size(); ++n)
+        {
+          const double va = vaVec.at(n);
+          const double vb = vbVec.at(n);
+          // Since there are no overflow/underflow bins, check the range
+          if(va < fBinsA.Min() || vb < fBinsB.Min()){ returnVec.push_back(-1); continue;}
+          if(va > fBinsA.Max() || vb > fBinsB.Max()){ returnVec.push_back(fBinsA.NBins() * fBinsB.NBins()); continue;}
+              
+          // FindBin uses root convention, first bin is bin 1, bin 0 is underflow
+          const int ia = fBinsA.FindBin(va) - 1;
+          const int ib = fBinsB.FindBin(vb) - 1;
+      
+          const int i = ia*fBinsB.NBins()+ib;
+              
+          returnVec.push_back(i+.5);
+        }
```

Comment:
Can you indent this according to the prevailing style?
```suggestion
      for(unsigned n = 0; n < vaVec.size(); ++n){
        const double va = vaVec.at(n);
        const double vb = vbVec.at(n);
        // Since there are no overflow/underflow bins, check the range
        if(va < fBinsA.Min() || vb < fBinsB.Min()){ returnVec.push_back(-1); continue;}
        if(va > fBinsA.Max() || vb > fBinsB.Max()){ returnVec.push_back(fBinsA.NBins() * fBinsB.NBins()); continue;}

        // FindBin uses root convention, first bin is bin 1, bin 0 is underflow
        const int ia = fBinsA.FindBin(va) - 1;
        const int ib = fBinsB.FindBin(vb) - 1;

        const int i = ia*fBinsB.NBins()+ib;

        returnVec.push_back(i+.5);
      }
```